erts/emulator/test/efile_SUITE.erl

```diff
@@ -162,3 +166,38 @@ open_files(Name) ->
             %		  io:format("Error reason: ~p", [_Reason]),
             []
     end.
+
+%% @doc If /proc filesystem exists (no way to know if it is real proc or just
+%% a /proc directory), let's read some zero sized files 500 times each, while
+%% ensuring that response isn't empty << >>
+proc_zero_sized_files(Config) when is_list(Config) ->
+    %% Some files which exist on Linux but might be missing on other systems
+    Inputs = ["/proc/cpuinfo",
+              "/proc/meminfo",
+              "/proc/partitions",
+              "/proc/swaps",
+              "/proc/version",
+              "/proc/uptime"],
+    case filelib:is_dir("/proc") of
+        false -> ok; % skip the test if no /proc
+        true ->
+            %% Take away files which do not exist in proc
+            Inputs1 = lists:filter(fun filelib:is_file/1, Inputs),
+
+            %% Fail if none of mentioned files exist in /proc, did we just get
+            %% a normal /proc directory without any special files?
+            ?assertNotEqual([], Inputs1),
```

Comment:
None of the /proc/ files in "Inputs" are available on freebsd or solaris.

On freebsd I suggest you use /proc/curproc/cmdline.
On solaris all of the files in /proc/ seem to have the correct filesize, so not sure what we can do there. I suggest skipping them.